uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Import/Parser/BaseClass.php

```diff
@@ -0,0 +1,323 @@
+      array_unshift($values, $this->leaveImportError[$params['absence_id']]);
+      return CRM_Import_Parser::ERROR;
+    }
+
+    $this->createBalanceChangeForLeaveDate($params, $this->leaveImportSuccess[$params['absence_id']]);
+    $transaction->commit();
+
+    return CRM_Import_Parser::VALID;
+  }
+
+  /**
+   * Gets available fields for import
+   *
+   * @return array
+   */
+  public static function getFields() {
+    return [
+      'contact_id' => [
+        'name' => 'contact_id',
+        'title' => ts('Contact ID'),
+        'type' => CRM_Utils_Type::T_INT,
+        'headerPattern' => '/Contact ID/i',
+      ],
+      'absence_id' => [
+        'name' => 'absence_id',
+        'title' => ts('Absence ID'),
+        'type' => CRM_Utils_Type::T_INT,
+        'headerPattern' => '/Absence ID/i',
+      ],
+      'absence_type' => [
+        'name' => 'absence_type',
+        'title' => ts('Absence Type'),
+        'type' => CRM_Utils_Type::T_STRING,
+        'headerPattern' => '/Absence Type/i',
+      ],
+      'absence_date' => [
+        'name' => 'absence_date',
+        'title' => ts('Absence Date'),
+        'type' => CRM_Utils_Type::T_DATE,
+        'headerPattern' => '/Absence Date/i',
+      ],
+      'qty' => [
+        'name' => 'qty',
+        'title' => ts('Qty'),
+        'type' => CRM_Utils_Type::T_FLOAT,
+        'headerPattern' => '/Qty/i',
+      ],
+      'start_date' => [
+        'name' => 'start_date',
+        'title' => ts('Start Date'),
+        'type' => CRM_Utils_Type::T_DATE,
+        'headerPattern' => '/Start Date/i',
+      ],
+      'end_date' => [
+        'name' => 'end_date',
+        'title' => ts('End Date'),
+        'type' => CRM_Utils_Type::T_DATE,
+        'headerPattern' => '/End Date/i',
+      ],
+      'total_qty' => [
+        'name' => 'total_qty',
+        'title' => ts('Total Qty'),
+        'type' => CRM_Utils_Type::T_FLOAT,
+        'headerPattern' => '/Total Qty/i',
+      ],
+      'status' => [
+        'name' => 'status',
+        'title' => ts('Status'),
+        'type' => CRM_Utils_Type::T_STRING,
+        'headerPattern' => '/Status/i',
+      ],
+      'comments' => [
+        'name' => 'comments',
+        'title' => ts('Comments'),
+        'type' => CRM_Utils_Type::T_STRING,
+        'headerPattern' => '/Comments/i',
+      ],
+    ];
+  }
+
+  /**
+   * Validate a row of values in CSV based on the field type
+   *
+   * @param array $params
+   *
+   * @return string
+   */
+  private function validateFieldTypes($params){
+    $errorMessage = NULL;
+    $fields = self::getFields();
+
+    foreach($params as $key=>$value) {
+      if (isset($fields[$key]['type']) && $fields[$key]['type'] == CRM_Utils_Type::T_DATE) {
+
+        try {
+          new DateTime($value);
+        } catch (Exception $e) {
+          CRM_Contact_Import_Parser_Contact::addToErrorMsg('Invalid Date value for '.$fields[$key]['title'], $errorMessage);
```

Comment:
Found the method used in all the Importers in core and in Job roles and Job contracts extension. just checked the method itself and it doesn't do much. Will just implement it in the parser file directly.